lib/worker.py

```diff
@@ -371,7 +373,13 @@ def run_loop(self, task_queue, result_queue):
                 short_status, duration = self.run_task(task_id)
                 retries_left = retries_left - 1
 
-            result_queue.put(self.wrap_result(task_id, short_status, duration))
+            if fails_count > 0 and short_status == 'pass':
+                result_queue.put(self.wrap_result(task_id, 'flaky', duration))
+            else:
+                result_queue.put(
+                    self.wrap_result(task_id, short_status, duration)
+                )
```

Comment:
I'm in doubt about encoding auxiliary information into the short status. `flaky` here means `passed after one or several fails`. It is `pass` in fact and any code that expects `pass` should also expect `flaky`. It looks error-prone.

We can extend the `WorkerTaskResult` class with one more field or even create a new class to deliver this information to the statistics watcher (so it'll be a separate message). Those ways looks more clean for me.

Reply:
Fixed.